File: src/algorithms/stacks-queues/validation/valid-parentheses/ValidParentheses_test.java
// javac ValidParentheses.java ValidParentheses_test.java && java -ea ValidParentheses_test
public class ValidParentheses_test {
    public static void main(String[] args) {
        assert ValidParentheses.validParentheses("({[]})") == true;
        assert ValidParentheses.validParentheses("()") == true;
        assert ValidParentheses.validParentheses("((()))") == true;
        assert ValidParentheses.validParentheses("()[]{}") == true;
        assert ValidParentheses.validParentheses("(]") == false;
        assert ValidParentheses.validParentheses("([)]") == false;
        assert ValidParentheses.validParentheses("(") == false;
        assert ValidParentheses.validParentheses(")") == false;
        assert ValidParentheses.validParentheses("") == true;
        assert ValidParentheses.validParentheses("({[]})(") == false;

        System.out.println("All tests passed!");
    }
}

File: src/algorithms/stacks-queues/validation/valid-parentheses/valid-parentheses_test.py
import importlib
import os
import sys

sys.path.insert(0, os.path.join(os.path.dirname(__file__), "sources"))

mod = importlib.import_module("valid-parentheses")
valid_parentheses = mod.valid_parentheses

assert valid_parentheses("({[]})") == True
assert valid_parentheses("()") == True
assert valid_parentheses("((()))") == True
assert valid_parentheses("()[]{}") == True
assert valid_parentheses("(]") == False
assert valid_parentheses("([)]") == False
assert valid_parentheses("(") == False
assert valid_parentheses(")") == False
assert valid_parentheses("") == True
assert valid_parentheses("({[]})(") == False

if __name__ == "__main__":
    print("All tests passed!")
